### PR TITLE
Fix #9844

### LINKED ISSUE #9844: object self-assignment order-of-evaluation broken
Same as #9684 but for objects...
test.nim
```nim
proc f(v: int): int = v

type X = object
  v: int

var x = X(v: 42)

x = X(v: f(x.v))
assert x.v == 42
```
```
bin/nim --version
Nim Compiler Version 0.19.9 [Linux: amd64]
Compiled at 2018-12-03
Copyright (c) 2006-2018 by Andreas Rumpf

git hash: c74226f4c1d156f2124783ac4464a8473d79c651
active boot switches: -d:release
```
```
bin/nim c -r ../../tmp/test
Hint: used config file '/home/arnetheduck/src/Nim/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: test [Processing]
CC: test
CC: stdlib_system
Hint:  [Link]
Hint: operation successful (12324 lines compiled; 0.412 sec total; 16.191MiB peakmem; Debug Build) [SuccessX]
Hint: /home/arnetheduck/tmp/test  [Exec]
test.nim(9)              test
system.nim(3928)         failedAssertImpl
system.nim(3921)         raiseAssert
system.nim(2968)         sysFatal
Error: unhandled exception: /home/arnetheduck/tmp/test.nim(9, 8) `x.v == 42`  [AssertionError]
Error: execution of an external program failed: '/home/arnetheduck/tmp/test '
```

### PR DESCRIPTION
**Pull request by Clyybber**
It should probably check for all accessed global symbols in the proc too?
Is there a helper proc that could give me a list of those symbols?

### CODE REVIEW

File: compiler/aliases.nim [hunk 1]
```diff
@@ -181,9 +181,16 @@ proc isPartOf*(a, b: PNode): TAnalysisResult =
       else: discard
     of nkObjConstr:
       result = arNo
-      for i in 1..<b.len:
+      for i in 1 .. <b.len:
```
Comment (kaushalmodi) [inline compiler/aliases.nim]: Isn't `<foo` deprecated?

Comment (kaushalmodi) [reply compiler/aliases.nim]: i.e isn't above same as `1 ..< b.len`? But that's the non-deprecated syntax?

Comment (Clyybber) [reply compiler/aliases.nim]: Ah, that's a typo. Thanks for checking :)

**Comment by cooldome** [comment]
Fantastic improvement, just replace `.. <` with `..<`

**Comment by Clyybber** [comment]
Apparently this also fixes #9684

**Comment by Clyybber** [comment]
Unrelated test failure on appveyor, because github's being blazingly fast again...

**Comment by Araq** [comment]
Note: Apparently no CIs did run?

**Pull request merged**